sdk/src/compressed_account.rs

```diff
@@ -1,421 +1,152 @@
 use std::ops::{Deref, DerefMut};
 
-use anchor_lang::prelude::{
-    AccountInfo, AnchorDeserialize, AnchorSerialize, ProgramError, Pubkey, Result,
-};
+use anchor_lang::prelude::{AnchorDeserialize, AnchorSerialize, ProgramError, Pubkey, Result};
 use light_hasher::{DataHasher, Discriminator, Hasher, Poseidon};
 use light_utils::hash_to_bn254_field_size_be;
+use solana_program::account_info::AccountInfo;
 
 use crate::{
-    address::{derive_address, NewAddressParamsPacked},
-    merkle_context::{
-        pack_merkle_context, MerkleContext, PackedAddressMerkleContext, PackedMerkleContext,
-        RemainingAccounts,
+    account_info::LightAccountInfo,
+    address::{
+        derive_address_from_params, derive_address_seed, unpack_new_address_params,
+        PackedNewAddressParams,
     },
-    program_merkle_context::unpack_address_merkle_context,
+    merkle_context::{pack_merkle_context, MerkleContext, PackedMerkleContext, RemainingAccounts},
 };
 
-pub trait LightAccounts: Sized {
-    fn try_light_accounts(
-        inputs: Vec<Vec<u8>>,
-        merkle_context: PackedMerkleContext,
-        merkle_tree_root_index: u16,
-        address_merkle_context: PackedAddressMerkleContext,
-        address_merkle_tree_root_index: u16,
-        remaining_accounts: &[AccountInfo],
-    ) -> Result<Self>;
-    fn new_address_params(&self) -> Vec<NewAddressParamsPacked>;
-    fn input_accounts(
-        &self,
-        remaining_accounts: &[AccountInfo],
-    ) -> Result<Vec<PackedCompressedAccountWithMerkleContext>>;
-    fn output_accounts(
-        &self,
-        remaining_accounts: &[AccountInfo],
-    ) -> Result<Vec<OutputCompressedAccountWithPackedContext>>;
+pub trait LightAccounts<'a>: Sized {
+    fn try_light_accounts(accounts: &'a [LightAccountInfo]) -> Result<Self>;
 }
 
+// TODO(vadorovsky): Implment `LightAccountLoader`.
+
 /// A wrapper which abstracts away the UTXO model.
-pub enum LightAccount<T>
+pub struct LightAccount<'info, T>
```

Comment:
Maybe we can add an anchor version of this which uses AnchorDeserialize and AnchorSerialize.
For example call it LightAnchorAccount or do feature gate:
```
#[cfg(feature(anchor)
AnchorDeserialize + AnchorSerialize
#[cfg(feature(anchor)
BorshDeserialize + BorshSerialize
```